projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts

```diff
@@ -51,20 +53,47 @@ export class NgSelectA11yDirective implements AfterViewInit {
     observer.observe(this.elementRef.nativeElement, { childList: true });
   }
 
+  /**
+   * When we inside a combo box using JAWS screen reader and press escape key
+   * an escape keyboard event doesn't get fired, instead an AltLeft is fired.
+   */
+  @HostListener('keyup', ['$event'])
+  onKeyUp(event: KeyboardEvent) {
+    if (
+      !this.featureConfigService?.isEnabled('a11yNgSelectCloseDropdownOnEscape')
+    ) {
+      return;
+    }
+    const jawsEscapeCode = 'AltLeft';
+    if (event.code === jawsEscapeCode) {
+      this.selectComponent.close();
+    }
+  }
+
   @Optional() breakpointService = inject(BreakpointService, { optional: true });
 
   @Inject(PLATFORM_ID) protected platformId: Object;
 
   constructor(
     private renderer: Renderer2,
-    private elementRef: ElementRef
+    private elementRef: ElementRef,
+    private selectComponent: NgSelectComponent
   ) {}
 
   ngAfterViewInit(): void {
     const divCombobox =
       this.elementRef.nativeElement.querySelector('[role="combobox"]');
     const inputElement = divCombobox.querySelector('input');
 
+    this.renderer.setAttribute(inputElement, 'role', 'combobox');
+    this.renderer.setAttribute(inputElement, 'aria-expanded', 'false');
+
+    const isOpened$ = this.selectComponent.openEvent.pipe(map(() => 'true'));
+    const isClosed$ = this.selectComponent.closeEvent.pipe(map(() => 'false'));
+    merge(isOpened$, isClosed$).subscribe((state) => {
```

Comment:
Lets tie this subscription to a parent subscription object that we can unsubscribe from on destroy to remove all our subsciptions at the end of this directive's life.

Reply:
@Zeyber fixed 👍